pkg/controller/bucket/bucket_controller.go

```diff
@@ -99,18 +100,37 @@ func (bl *bucketListener) Add(ctx context.Context, obj *v1alpha1.Bucket) error {
 	}
 
 	req := osspec.ProvisionerCreateBucketRequest{
-		BucketName: obj.Spec.BucketRequest.Name,
+		BucketContext: map[string]string{},
+	}
+	if len(obj.Spec.Parameters["BucketInstanceName"]) > 0 {
+		req.BucketName = obj.Spec.Parameters["BucketInstanceName"]
+	} else if len(obj.Spec.Parameters["BucketPrefix"]) > 0 {
+		req.BucketContext["BucketPrefix"] = obj.Spec.Parameters["BucketPrefix"]
+	}
+
+	if obj.Spec.AnonymousAccessMode.Private {
+		req.AnonymousBucketAccessMode = osspec.ProvisionerCreateBucketRequest_BUCKET_PRIVATE
+	} else if obj.Spec.AnonymousAccessMode.PublicReadOnly {
+		req.AnonymousBucketAccessMode = osspec.ProvisionerCreateBucketRequest_BUCKET_READ_ONLY
+	} else if obj.Spec.AnonymousAccessMode.PublicReadWrite {
+		req.AnonymousBucketAccessMode = osspec.ProvisionerCreateBucketRequest_BUCKET_WRITE_ONLY
+	} else if obj.Spec.AnonymousAccessMode.PublicWriteOnly {
+		req.AnonymousBucketAccessMode = osspec.ProvisionerCreateBucketRequest_BUCKET_READ_WRITE
```

Comment:
can all of these constants be put in a map such that a single stmt gets us the value? Eg:
```
     req.AnonymousBucketAccessMode = anonAccessMap[obj.Spec.AnonymousAccessMode]
```